integration-tests/test_registration.py

```diff
@@ -0,0 +1,58 @@
+import os
+import subprocess
+
+import pytest
+
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+MACHINE_ID_FILE: str = "/etc/insights-client/machine-id"
+
+
+def test_machineid_exists_only_when_registered(insights_client):
+    """`machine-id` is only present when insights-client is registered."""
+    assert not insights_client.is_registered
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+    res = insights_client.run(check=False)
+    assert (
+        "This host has not been registered. Use --register to register this host."
+    ) in res.stdout
+    assert res.returncode != 0
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+    insights_client.register()
+    assert os.path.exists(MACHINE_ID_FILE)
+
+    insights_client.unregister()
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+
+def test_machineid_changes_on_new_registration(insights_client):
+    """machine-id content changes when insights-client is un- & registered."""
+    insights_client.register()
+    with open(MACHINE_ID_FILE, "r") as f:
+        machine_id_old = f.read()
+
+    insights_client.unregister()
+    assert not os.path.exists(MACHINE_ID_FILE)
+
+    insights_client.register()
+    with open(MACHINE_ID_FILE, "r") as f:
+        machine_id_new = f.read()
+
+    assert machine_id_new != machine_id_old
+
+
+def test_double_register(insights_client):
+    """`--register` can be passed multiple times.
```

Comment:
I'd expand the description here a bit, to be more explicit on what happens; something like:
```
    `--register` can be invoked without failures also on an already
    registered system
```